### PR TITLE
chore: readme update

### PR DESCRIPTION
**Pull request by MoumitaM**
# Description

> Readme file updated with latest JS SDK loading snippet.

# Linear ticket
[Click here](https://linear.app/rudderstack/issue/SDK-2338/update-ruddertyper-readme-file)

# Thank You for Contributing to RudderTyper!

We sincerely appreciate the time and effort you invest in helping to improve RudderTyper. To make the most of your contribution, we kindly ask that you document your Pull Request thoroughly.

Please note that if the information provided is insufficient, or if the Pull Request does not align with our roadmap, we may need to respectfully thank you for your effort and close the issue.

## Description 📝

> Please provide a detailed description of the changes you’ve made in this section. Clearly explain what has been modified, added, or removed, and how it impacts the overall project.

## Respect Earns Respect 👏

We ask that you adhere to our [Code of Conduct](CODE_OF_CONDUCT.md). In summary:

- Use welcoming and inclusive language.
- Respect differing viewpoints and experiences.
- Accept constructive criticism gracefully.
- Focus on what is best for the community.
- Show empathy toward other community members.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -277,6 +377,8 @@ browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
 </script>
 ```
 
+> Note: Remember to replace all `WRITE_KEY` and `DATA_PLANE_URL` with your write key and data plane url.
```
Comment (saikumarrs) [inline README.md:380]:
```suggestion
> Note: Remember to replace all instances of `WRITE_KEY` and `DATA_PLANE_URL` in the snippet with your write key and data plane URL values respectively.
```

File: README.md [hunk 2]
```diff
@@ -227,39 +227,139 @@ rudderTyper.orderCompleted({
...
+
+For typescript project add `tsify` npm package as dev dependency and run the following command:
+
+```sh
+npx browserify analytics/index.js -p [ tsify ] --standalone rudderTyper >  rudderTyperBundle.js
 ```
 
 - Now you can make calls from your `html` file as shown:
 
 ```html
```
Comment (saikumarrs) [inline README.md:241]: Let's not add the RudderStack snippet here. Please update the instructions to point to the specific docs section.

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (MoumitaM) [reply README.md:224]: Addressed

File: README.md (hunk 1 above)
Comment (MoumitaM) [reply README.md]: Done

File: README.md [hunk 3]
```diff
@@ -196,76 +196,33 @@ import com.rudderstack.generated.*
 
 ### RudderStack JavaScript SDK
 
-- Import the RudderTyper-generated client using `require()` and make the calls if your framework supports them. Otherwise, you can use [**Browserify**](https://browserify.org/) to generate a bundle that supports your implementation. The implementation for each of the alternatives mentioned above will be as shown:
+There are two ways to get started with Rudder-Typer in your browser:
 
-#### Using the `require()`method
+#### 1. Using CDN (Snippet)
```
Comment (saikumarrs) [inline README.md:201]: Can we also point in the instructions at the end to the appropriate sample application for more details?

File: README.md [hunk 4]
```diff
@@ -196,76 +196,33 @@ import com.rudderstack.generated.*
...
-  total: 39.99,
-});
+```sh
+npx browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
 ```
 
-#### Using `browserify`
-
-- Execute the following command to generate a bundle from the RudderTyper client:
+- For typescript project add `tsify` npm package as dev dependency and run the following command:
```
Comment (saikumarrs) [inline README.md:216]:
```suggestion
- For TypeScript analytics client, add `tsify` NPM package as a dev dependency and run the following command to generate the bundle:
```

File: README.md [hunk 5]
```diff
@@ -196,76 +196,33 @@ import com.rudderstack.generated.*
...
 
 ```sh
-browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
+npx browserify analytics/index.js -p [ tsify ] --standalone rudderTyper >  rudderTyperBundle.js
```
Comment (saikumarrs) [inline README.md:219]: This is the main change in the command from the previous instruction. Please be a little more cautious.
```suggestion
npx browserify analytics/index.ts -p [ tsify ] --standalone rudderTyper >  rudderTyperBundle.js
```

File: README.md [hunk 6]
```diff
@@ -196,76 +196,33 @@ import com.rudderstack.generated.*
...
-    rudderanalytics.page();
-  </script>
-  <script src="https://cdn.rudderlabs.com/v1/rudder-analytics.min.js"></script>
-  <script src="./rudderTyperBundle.js"></script>
-  <meta charset="UTF-8" />
-  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
-  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-  <title>Document</title>
-</head>
+<script src="./rudderTyperBundle.js"></script>
```
Comment (saikumarrs) [inline README.md:225]: What should be the order of the installation of the SDK snippet and this bundle?

Can you clarify this with an example without using the actual snippet?

File: README.md [hunk 7]
```diff
@@ -277,6 +234,35 @@ browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
...
+});
+
+// Issue your first RudderTyper track call!
+RudderTyperAnalytics.orderCompleted({
+  orderID: 'ck-f306fe0e-cc21-445a-9caa-08245a9aa52c',
+  total: 39.99,
+});
+```
+
+> Note: Remember to replace all instances of `WRITE_KEY` and `DATA_PLANE_URL` in the snippet with your write key and data plane URL values respectively.
```
Comment (saikumarrs) [inline README.md:262]: Please move this to the CDN section.

Comment (saikumarrs) [reply README.md:262]: If they use the snippet from the RS dashboard, these values will be prefilled. So, no need for this instruction itself. Apologies if your original intention was the same.

If you think it'll be helpful for the NPM case, keep it but phrase it appropriately.

File: README.md (hunk 3 above)
Comment (saikumarrs) [reply README.md:201]: I felt I didn't have to point this out. Let's do the same for NPM.

Comment (MoumitaM) [reply README.md:201]: Done

File: README.md (hunk 7 above)
Comment (MoumitaM) [reply README.md]: Done

**Review by saikumarrs — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -196,76 +196,36 @@ import com.rudderstack.generated.*
 
 ### RudderStack JavaScript SDK
 
-- Import the RudderTyper-generated client using `require()` and make the calls if your framework supports them. Otherwise, you can use [**Browserify**](https://browserify.org/) to generate a bundle that supports your implementation. The implementation for each of the alternatives mentioned above will be as shown:
+There are two ways to get started with Rudder-Typer in your browser:
```
Comment (ameypv-rudder) [inline README.md:199]:
```suggestion
There are two ways to get started with RudderTyper in your browser:
```

File: README.md [hunk 9]
```diff
@@ -196,76 +196,36 @@ import com.rudderstack.generated.*
...
 
-#### Using the `require()`method
+#### 1. Using CDN (Snippet)
```
Comment (ameypv-rudder) [inline README.md:201]:
```suggestion
#### 1. Using the JavaScript SDK snippet
```

File: README.md [hunk 10]
```diff
@@ -196,76 +196,36 @@ import com.rudderstack.generated.*
...
 
-```javascript
-// Import RudderStack JS SDK and initialize it
-const RudderAnalytics = require('@rudderstack/analytics-js');
-
-const rudderAnalytics = new RudderAnalytics();
-rudderAnalytics.load(WRITE_KEY, DATA_PLANE_URL, {});
+- Paste the snippet from the Rudderstack dashboard to your HTML file. [Ref](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/quickstart/#using-cdn)
```
Comment (ameypv-rudder) [inline README.md:203]:
```suggestion
- Paste the [JavaScript SDK snippet](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/quickstart/#using-cdn) from the RudderStack dashboard to your HTML file.
```

File: README.md [hunk 11]
```diff
@@ -196,76 +196,36 @@ import com.rudderstack.generated.*
...
-const rudderTyper = require('./analytics/index');
+```sh
+npm install --save-dev  @rudderstack/analytics-js
+```
 
-// Pass in your @rudderstack/analytics-js instance to RudderTyper client
-rudderTyper.setRudderTyperOptions({
-  analytics: rudderAnalytics,
-});
+- Execute the following command to generate a bundle from the RudderTyper client:
```
Comment (ameypv-rudder) [inline README.md:210]:
```suggestion
- Run the following command to generate a bundle from the RudderTyper client:
```

File: README.md [hunk 12]
```diff
@@ -196,76 +196,36 @@ import com.rudderstack.generated.*
...
-  total: 39.99,
-});
+```sh
+npx browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
 ```
 
-#### Using `browserify`
-
-- Execute the following command to generate a bundle from the RudderTyper client:
+- For TypeScript analytics client, add `tsify` NPM package as a dev dependency and run the following command to generate the bundle:
```
Comment (ameypv-rudder) [inline README.md:216]:
```suggestion
- For the TypeScript analytics client, add the NPM package `tsify` as a dependency and run the following command to generate the bundle:
```

File: README.md [hunk 13]
```diff
@@ -196,76 +196,36 @@ import com.rudderstack.generated.*
...
 
 ```sh
-browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
+npx browserify analytics/index.ts -p [ tsify ] --standalone rudderTyper >  rudderTyperBundle.js
 ```
 
-- Now you can make calls from your `html` file as shown:
+- Now import your RudderTyper client and send events
```
Comment (ameypv-rudder) [inline README.md:222]:
```suggestion
- Import your RudderTyper client and send events, as shown:
```

File: README.md [hunk 14]
```diff
@@ -277,6 +237,39 @@ browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
 </script>
 ```
 
+> Check out the [sample application](https://github.com/rudderlabs/rudder-typer/tree/develop/examples/js-cdn-typescript) for reference.
```
Comment (ameypv-rudder) [inline README.md:240]:
```suggestion
> See the [sample application](https://github.com/rudderlabs/rudder-typer/tree/develop/examples/js-cdn-typescript) for reference.
```

File: README.md [hunk 15]
```diff
@@ -277,6 +237,39 @@ browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
...
+  orderID: 'ck-f306fe0e-cc21-445a-9caa-08245a9aa52c',
+  total: 39.99,
+});
+```
+
+> Note: Remember to replace all instances of `WRITE_KEY` and `DATA_PLANE_URL` with your write key and data plane URL values respectively.
+
+> Note: Run `npx rudder-typer` to regenerate your RudderTyper client. You need to do this each time you update your Tracking Plan.
+
+> Check out the [sample application](https://github.com/rudderlabs/rudder-typer/tree/develop/examples/js-npm-typescript) for reference.
```
Comment (ameypv-rudder) [inline README.md:271]:
```suggestion
See the [sample application](https://github.com/rudderlabs/rudder-typer/tree/develop/examples/js-npm-typescript) for reference.
```

File: README.md [hunk 16]
```diff
@@ -277,6 +237,39 @@ browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
...
+});
+
+// Issue your first RudderTyper track call!
+RudderTyperAnalytics.orderCompleted({
+  orderID: 'ck-f306fe0e-cc21-445a-9caa-08245a9aa52c',
+  total: 39.99,
+});
+```
+
+> Note: Remember to replace all instances of `WRITE_KEY` and `DATA_PLANE_URL` with your write key and data plane URL values respectively.
```
Comment (ameypv-rudder) [inline README.md:267]:
```suggestion
Note the following: 

- Remember to replace `WRITE_KEY` and `DATA_PLANE_URL` in the above snippet with the write key of your JavaScript source and the data plane URL respectively.
- Run `npx rudder-typer` to regenerate your RudderTyper client every time you update your tracking plan.
```

File: README.md [hunk 17]
```diff
@@ -277,6 +237,39 @@ browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
...
+
+> Note: Run `npx rudder-typer` to regenerate your RudderTyper client. You need to do this each time you update your Tracking Plan.
```
Comment (ameypv-rudder) [inline README.md:269]:
```suggestion
```

File: README.md [hunk 18]
```diff
@@ -277,6 +237,39 @@ browserify analytics/index.js --standalone rudderTyper >  rudderTyperBundle.js
...
+
+// Issue your first RudderTyper track call!
+RudderTyperAnalytics.orderCompleted({
+  orderID: 'ck-f306fe0e-cc21-445a-9caa-08245a9aa52c',
+  total: 39.99,
+});
+```
+
+> Note: Remember to replace all instances of `WRITE_KEY` and `DATA_PLANE_URL` with your write key and data plane URL values respectively.
+
```
Comment (ameypv-rudder) [inline README.md:268]:
```suggestion
```

**Review by ameypv-rudder — APPROVED** [review]
Made some minor suggestions - LGTM otherwise.

**Pull request merged**